src/user/UserCredentials.tsx

```diff
@@ -205,17 +370,42 @@ export const UserCredentials = ({ user }: UserCredentialsProps) => {
                     fieldLabelId="users:userCredentialsHelpTextLabel"
                   />
                 </Th>
+                <Th />
                 <Th>{t("type")}</Th>
                 <Th>{t("userLabel")}</Th>
                 <Th>{t("data")}</Th>
                 <Th />
                 <Th />
               </Tr>
             </Thead>
-            <Tbody>
+            <Tbody
+              ref={bodyRef}
+              onDragOver={onDragOver}
+              onDrop={onDragOver}
+              onDragLeave={onDragLeave}
+            >
               {groupedUserCredentials.map((groupedCredential, rowIndex) => (
                 <Fragment key={`table-${groupedCredential.key}`}>
-                  <Tr>
+                  <Tr
+                    id={groupedCredential.value
+                      .map((credential) => {
+                        return credential.id;
+                      })
+                      .toString()}
+                    draggable
+                    onDrop={onDrop}
+                    onDragEnd={onDragEnd}
+                    onDragStart={onDragStart}
+                  >
+                    <Td
+                      draggableRow={{
+                        id: `draggable-row-${groupedCredential.value.map(
+                          (credential) => {
+                            return credential.id;
+                          }
```

Comment:
This code could be a little smaller.
```suggestion
                        id: `draggable-row-${groupedCredential.value.map(
                          ({ id }) => id
                        )}`,
```